src/views/Nft/market/hooks/useNftsForAddress.tsx

```diff
@@ -26,12 +26,10 @@ const useNftsForAddress = (account: string, profile: Profile, isProfileFetching:
     return null
   }, [profileNftTokenId, profileNftCollectionAddress, hasProfileNft])
 
-  const { status, data, mutate } = useSWR([account, collections, isProfileFetching, 'userNfts'], async () => {
-    if (!isProfileFetching && !isEmpty(collections) && isAddress(account)) {
-      return getCompleteAccountNftData(account, collections, profileNftWithCollectionAddress)
-    }
-    return []
-  })
+  const { status, data, mutate } = useSWR(
+    !isProfileFetching && !isEmpty(collections) && isAddress(account) ? [account, 'userNfts'] : null,
+    async () => getCompleteAccountNftData(account, collections, profileNftWithCollectionAddress),
```

Comment:
should we consider `[account, collections, profileNftWithCollectionAddress, 'userNfts']` as key to revalidate everytime those 4 variables change?